libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1148,39 +1136,41 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
     /**
      * Process the initial request to establish a long lived connection via a streaming server.
      * @param req The connection request.
-     * @param res The response sent on error or connection termination.
-     * @param logic The logic that will handle incoming requests.
+     * @param socket The raw socket connection between the bot (server) and channel/caller (client).
+     * @param head The first packet of the upgraded stream.
+     * @param logic The logic that handles incoming streaming requests for the lifetime of the WebSocket connection.
      */
-    public async useWebSocket(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {   
+    public async useWebSocket(req: IncomingMessage, socket: Socket, head: Buffer, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+        // Use the provided NodeWebSocketFactoryBase on BotFrameworkAdapter construction,
+        // otherwise create a new NodeWebSocketFactory.
+        const webSocketFactory = this.webSocketFactory || new NodeWebSocketFactory();
+
         if (!logic) {
             throw new Error('Streaming logic needs to be provided to `useWebSocket`');
         }
 
-        if (!this.webSocketFactory || !this.webSocketFactory.createWebSocket) {
-            throw new Error('BotFrameworkAdapter must have a WebSocketFactory in order to support streaming.');
-        }
-
         this.logic = logic;
 
-        // Restify-specific check.
-        if (typeof((res as any).claimUpgrade) !== 'function') {
-            throw new Error('ClaimUpgrade is required for creating WebSocket connection.');
-        }
-
         try {
             await this.authenticateConnection(req, this.settings.channelService);
         } catch (err) {
-            // Set the correct status code for the socket to send back to the channel.
-            res.status(StatusCodes.UNAUTHORIZED);
-            res.send(err.message);
+            // If the authenticateConnection call fails, send back the correct error code and close
+            // the connection.
+            if (typeof(err.message) === 'string' && err.message.toLowerCase().startsWith('unauthorized')) {
+                abortWebSocketUpgrade(socket, 401);
+            } else if (typeof(err.message) === 'string' && err.message.toLowerCase().startsWith(`'authheader'`)) {
+                abortWebSocketUpgrade(socket, 400);
+            } else {
+                abortWebSocketUpgrade(socket, 500);
```

Comment:
Why dont we send the error message in the response? Seems impossible to detect the error without some more info.

Reply:
Or are they usually cryptic?